File: src/main/kotlin/at/searles/parsing/tools/common/ToInt.java
package at.searles.parsing.tools.common;

import at.searles.parsing.Mapping;
import at.searles.parsing.ParserStream;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class ToInt implements Mapping<CharSequence, Integer> {
    @Override
    public Integer parse(ParserStream stream, @NotNull CharSequence left) {
        try {
            return Integer.parseInt(left.toString());
        } catch (NumberFormatException e) {
            return null;
        }
    }

    @Nullable
    @Override
    public CharSequence left(@NotNull Integer result) {
        return result.toString();
    }
}

File: src/main/kotlin/at/searles/parsingtools/common/ToInt.kt
package at.searles.parsingtools.common

import at.searles.parsing.Mapping
import at.searles.parsing.ParserStream

class ToInt : Mapping<CharSequence, Int> {
    override fun parse(stream: ParserStream, left: CharSequence): Int? {
        try {
            return Integer.parseInt(left.toString())
        } catch (e: NumberFormatException) {
            return null
        }

    }

    override fun left(result: Int): CharSequence? {
        return result.toString()
    }
}
